src/service/CinemaService.java

```diff
@@ -0,0 +1,179 @@
+                case "0" -> {
+                    return;
+                }
+                case "1" -> showSeats();
+                case "2" -> buyTicket();
+                case "3" -> statistics();
+                default -> {
+                    System.out.printf("Unknown menu item - %s\n", input);
+                    showMenu();
+                }
+            }
+        }
+    }
+
+    private void showSeats() {
+        System.out.print("Cinema:\n ");
+        for (int i = 1; i <= cinema.getSeats()[0].length; i++) {
+            System.out.print(" " + i);
+        }
+
+        System.out.println();
+
+        int counter = 1;
+        for (String[] seat : cinema.getSeats()) {
+            System.out.print(counter);
+            for (String s : seat) {
+                System.out.print(" " + s);
+            }
+            System.out.println();
+            counter++;
+        }
+
+        System.out.println();
+    }
+
+    private void buyTicket() {
+        try {
+            Cinema.Seat seat = seatValidation();
+            int price = getPrice(seat);
+
+            cinema.setCurrentIncome(cinema.getCurrentIncome() + price);
+            cinema.setSoldTickets(cinema.getSoldTickets() + 1);
+
+            printPrice(price);
+        } catch (AlreadyTakenSeatException e) {
+            System.out.println("That ticket has already been purchased!\n");
+            buyTicket();
+        } catch (NonExistingSeatException e) {
+            System.out.println("Wrong input!\n");
+            buyTicket();
+        }
+    }
+
+    private void statistics() {
+        System.out.printf("""
+                Number of purchased tickets: %d
+                Percentage: %s%%
+                Current income: $%d
+                Total income: $%d
+                                      
+                """, cinema.getSoldTickets(), getPercentage(), cinema.getCurrentIncome(), getTotalIncome());
+    }
+
+    private Cinema.Seat seatValidation() throws NonExistingSeatException, AlreadyTakenSeatException {
+        Scanner scanner = new Scanner(System.in);
+
+        System.out.println("Enter a row number:");
+        int row = scanner.nextInt();
+
+        System.out.println("Enter a seat number in that row:");
+        int column = scanner.nextInt();
+
+        System.out.println();
+
+        if (row < 1 || row > cinema.getRows() || column < 1 || column > cinema.getColumns()) {
+            throw new NonExistingSeatException();
+        }
+
+        if (cinema.getSeats()[row - 1][column - 1].equals("B")) {
+            throw new AlreadyTakenSeatException();
+        } else {
+            cinema.getSeats()[row - 1][column - 1] = "B";
+        }
+
+        return new Cinema.Seat(row, column);
+    }
+
+    private int getTotalSeats() {
+        return cinema.getColumns() * cinema.getRows();
+    }
+
+    private int getPrice(Cinema.Seat seat) {
+        int price;
+        boolean isFrontSide = seat.row <= cinema.getRows() / 2;
```

Comment:
Can you use the access method to access `seat.row`?

Reply:
Created getters and used them instead of direct field invocation